ethereum-fee-prepay/src/lib.rs

```diff
@@ -46,63 +48,40 @@ pub trait EthereumFeePrepay {
-        priority: Priority,
-    ) -> SCResult<()> {
+    fn pay_fee(&self, tx_senders: Vec<Address>, relayer: Address) -> SCResult<()> {
         self.require_whitelisted()?;
 
-        let estimate = self.compute_estimate(transaction_type, priority)?;
-        self.transfer(&address, &relayer, &estimate);
+        // To save gas for the relayers, we always use the latest queried value
+        // No need to check for empty, since this is guaranteed to be set by a previous CreateTransaction in EsdtSafe
+        let estimate = self.last_query_price().get() * ETH_ERC20_TX_GAS_LIMIT.into();
+        for tx_sender in tx_senders {
+            self.transfer(&tx_sender, &relayer, &estimate);
+        }
 
         Ok(())
     }
 
     #[endpoint(reserveFee)]
-    fn reserve_fee(
-        &self,
-        address: Address,
-        transaction_type: TransactionType,
-        priority: Priority,
-    ) -> SCResult<()> {
+    fn reserve_fee(&self, address: Address) -> SCResult<()> {
         self.require_whitelisted()?;
 
-        let estimate = self.compute_estimate(transaction_type, priority)?;
+        let estimate = self.compute_estimate();
         self.try_reserve_from_balance(&address, &estimate)?;
 
         Ok(())
     }
 
     #[endpoint(computeEstimate)]
-    fn compute_estimate(
-        &self,
-        transaction_type: TransactionType,
-        priority: Priority,
-    ) -> SCResult<Self::BigUint> {
-        let optional_arg_round = self
+    fn compute_estimate(&self) -> Self::BigUint {
+        let aggregator_result: AggregatorResult<Self::BigUint> = self
             .aggregator_proxy(self.aggregator().get())
-            .get_latest_round_data()
-            .execute_on_dest_context();
-        let aggregator_result: AggregatorResult<Self::BigUint> =
-            aggregator_result::try_parse_round(optional_arg_round)?;
-
-        let gas_limit = match transaction_type {
-            TransactionType::Ethereum => aggregator_result.transaction_gas_limits.ethereum,
-            TransactionType::Erc20 => aggregator_result.transaction_gas_limits.erc20,
-            TransactionType::Erc721 => aggregator_result.transaction_gas_limits.erc721,
-            TransactionType::Erc1155 => aggregator_result.transaction_gas_limits.erc1155,
-        };
-        let gas_price = match priority {
-            Priority::Fast => aggregator_result.priority_gas_costs.fast,
-            Priority::Average => aggregator_result.priority_gas_costs.average,
-            Priority::Low => aggregator_result.priority_gas_costs.low,
-        };
+            .latest_price_feed(BoxedBytes::from(GWEI_STRING), BoxedBytes::from(EGLD_STRING))
+            .execute_on_dest_context()
+            .into();
 
-        Ok(aggregator_result.egld_to_eth * gas_limit * gas_price
-            / aggregator_result.egld_to_eth_scaling)
+        self.last_query_price().set(&aggregator_result.price);
+
+        aggregator_result.price * ETH_ERC20_TX_GAS_LIMIT.into()
```

Comment:
i don't think this is correct. multiplying gas limit with a 18 decimal based denomination (EGLD) will result in precision errors. I'm not sure how this is intended to work, but if you wish to estimate the tx cost in Ethereum, using the GWEI to EGLD denomination might not be the way to go.

Reply:
tested and the relative value in USD is equivalent. closing.